src/FixerConfiguration/FixerConfigurationResolver.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\FixerConfiguration;
+
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class FixerConfigurationResolver implements FixerConfigurationResolverInterface
+{
+    /**
+     * @var array<string, FixerOptionInterface>
+     */
+    private $options = array();
+
+    /**
+     * @var string|null
+     */
+    private $root;
+
+    /**
+     * @param FixerOptionInterface $option
+     *
+     * @throws \LogicException when the option is already defined
+     *
+     * @return $this
+     */
+    public function addOption(FixerOptionInterface $option)
```

Comment:
Having this method can be useful when a set of fixers share common options but some of them have specific ones too: the common options can be defined in an abstract parent class and the child can still add new options. I'd like to use that in #2220 :)